proposals/20230316-oci-content-source.md

```diff
@@ -0,0 +1,284 @@
+- [Bundled Component and Bundling](https://github.com/bytecodealliance/SIG-Registries/blob/main/glossary.md#bundled-component-and-bundling): A "bundled component" is a [component](https://github.com/bytecodealliance/SIG-Registries/blob/main/glossary.md#component) that only has interface dependencies and can thus run directly on a wasm engine that natively implements those interfaces
+without requiring any registry access. "Bundling" is an automatic transformation on a [component](https://github.com/bytecodealliance/SIG-Registries/blob/main/glossary.md#component) that replaces [imports](https://github.com/bytecodealliance/SIG-Registries/blob/main/glossary.md#imports) of other components (in the
+registry) with inline copies of those components (fetched from the registry at the time of bundling) to produce a bundled component.
+
+## Proposal
+### User Stories
+#### Story 1 - Publishing a Component
+Alex is an engineer working in a large organization which is building applications using Wasm components. Alex would like share the component that their team has built with others within their company. The company Alex works for has a lot of folks with experience running containers, and they have OCI registries already provisioned to store container images. Alex would like to publish the component their team has built into one of their company's OCI registry.
+
+#### Story 2 - Publishing a Component Interface
+Erin is an engineer working on an open source project that allows users of the project to extend the functionality of the project using Wasm components as plugins. Erin would like to publish the interface specification for their plugin, so that other developers can easily find the interface and develop plugins for their project. Erin would like to use the GitHub Container Registry to store the Wasm component interface.
+
+#### Story 3 - Publishing a Bundled Component
+Alex is an engineer working in a large, security focused organization which runs a lot of Linux containers in production. The security team at Alex's company requires Linux containers be signed and provide a software bill of materials. Alex has recently built a new application that instead of being packaged as a Linux container image, they have built their application targeting Wasm. In fact, Alex built their application using many Wasm components. Alex and their team have finalized the feature set for their first release, tested the application, and locked the version for all the dependencies. Alex would like to publish this version of their application with all the application dependencies bundled together. Alex would also like to sign the bundled application and includes a software bill of materials of the components bundled. 
+
+#### Story 4 - Fetching a Component
```

Comment:
Is there a story for how these components are eventually run?  How are they to be consumed after they are pulled down?  

Is there a chance that there are platform specifics bits that are required to run the modules (I was looking at the image for slight which seems to need certs [installed along side the module](https://github.com/deislabs/containerd-wasm-shims/blob/a662b071189e13d2de17d0226e7c498f62052cd7/images/slight/Dockerfile#L5))

Reply:
I created an issue here: https://github.com/deislabs/containerd-wasm-shims/issues/89